test/exprtest/blas1_axpy_copy_test.cpp

```diff
@@ -0,0 +1,143 @@
+
+  int size;
+  scalar_t alpha;
+  int incX;
+  int incY;
+  std::tie(size, alpha, incX, incY) = combi;
+
+  // Dimensions of input vectors x and y
+  int x_dim = size * incX;
+  int y_dim = size * incY;
+
+  // Input vectors x and y (y is also output of AXPY)
+  std::vector<scalar_t> v_x(x_dim);
+  std::vector<scalar_t> v_y(y_dim);
+  fill_random(v_x);
+  fill_random(v_y);
+
+  // Copy of vector x
+  std::vector<scalar_t> v_xcopy(x_dim);
+
+  // Copy of vector y
+  // Used to store output on host for later comparison with SYCL result
+  std::vector<scalar_t> v_cpu_y(y_dim);
+
+  // Reference BLAS implementation
+  reference_blas::copy(size, v_x.data(), incX, v_xcopy.data(), incX);
+  reference_blas::copy(y_dim, v_y.data(), 1, v_cpu_y.data(), 1);
+  reference_blas::axpy(size, alpha, v_xcopy.data(), incX, v_cpu_y.data(), incY);
+
+  // SYCL-BLAS implementation
+  SYCL_DEVICE_SELECTOR d;
+  auto q = blas_test_t::make_queue(d);
+  Executor<executor_t> ex(q);
+
+  // Iterators
+  auto gpu_x_v = blas::make_sycl_iterator_buffer<scalar_t>(x_dim);
+  auto gpu_y_v = blas::make_sycl_iterator_buffer<scalar_t>(y_dim);
+  auto gpu_xcopy_v = blas::make_sycl_iterator_buffer<scalar_t>(x_dim);
+  auto gpu_ycopy_v = blas::make_sycl_iterator_buffer<scalar_t>(y_dim);
+
+  // Copy input data from host to device
+  auto xcp_ev =
+      ex.get_policy_handler().copy_to_device(v_x.data(), gpu_x_v, x_dim);
+  ex.get_policy_handler().wait(xcp_ev);
+  auto ycp_ev =
+      ex.get_policy_handler().copy_to_device(v_y.data(), gpu_y_v, y_dim);
+  ex.get_policy_handler().wait(ycp_ev);
+
+  // Dimensions of vector view for AXPY operations
+  int view_x_dim = (x_dim + incX - 1) / incX;
+  int view_y_dim = (y_dim + incY - 1) / incY;
+
+  // Views
+  // - for axpy operation where arbitrary stride is used
+  auto view_x_incX = make_vector_view(ex, gpu_x_v, incX, view_x_dim);
+  auto view_y_incY = make_vector_view(ex, gpu_y_v, incY, view_y_dim);
+
+  // - for copy operations where we want stride = 1
+  auto view_xcopy_inc1 = make_vector_view(ex, gpu_xcopy_v, 1, x_dim);
+  auto view_ycopy_inc1 = make_vector_view(ex, gpu_ycopy_v, 1, y_dim);
+
+  // Expressions to copy from device to device
+  auto xCopyOp = make_op<Assign>(view_xcopy_inc1, view_x_incX);
+  auto yCopyOp = make_op<Assign>(view_ycopy_inc1, view_y_incY);
+
+  // AXPY expressions
+  auto axpy_scal_op = make_op<ScalarOp, ProductOperator>(alpha, xCopyOp);
+  auto axpy_add_op = make_op<BinaryOp, AddOperator>(yCopyOp, axpy_scal_op);
+  auto copy_axpy_op_tree = make_op<Assign>(view_y_incY, axpy_add_op);
+
+  // Execute the COPY+AXPY tree
+  auto axpy_event = ex.execute(copy_axpy_op_tree);
+  ex.get_policy_handler().wait(axpy_event);
+
+  // Copy the result back to host memory
+  auto getResultEv =
+      ex.get_policy_handler().copy_to_host(gpu_y_v, v_y.data(), y_dim);
+  ex.get_policy_handler().wait(getResultEv);
+
+  ASSERT_TRUE(utils::compare_vectors(v_cpu_y, v_y));
+}
+
+#ifdef STRESS_TESTING
```

Comment:
The stress testing is used for application like gemm which takes too long to execute. If the stress testing for you takes less that two second you don't need to create stress testing